pico_limbo/src/server_state.rs

```diff
@@ -36,6 +36,12 @@ pub enum Boundaries {
     },
 }
 
+#[derive(Default)]
+pub struct TabList {
+    pub header: Option<Component>,
+    pub footer: Option<Component>,
+}
```

Comment:
I would suggest using a fat enum instead of a struct so that all the states can be represented:

```suggestion
#[derive(Default)]
enum TabList {
  #[default]
  None,
  Header { header: Component },
  Footer { footer: Component },
  HeaderAndFooter { header: Component, footer: Component },
}
```